adapters/insticator/insticator.go

```diff
@@ -0,0 +1,526 @@
+	StartDelay     *int     `json:"startdelay"`
+	Linearity      *int     `json:"linearity"`
+	Skip           *int     `json:"skip"`
+	SkipMin        *int     `json:"skipmin"`
+	SkipAfter      *int     `json:"skipafter"`
+	Sequence       *int     `json:"sequence"`
+	Battr          []int    `json:"battr"`
+	MaxExtended    *int     `json:"maxextended"`
+	MinBitrate     *int     `json:"minbitrate"`
+	MaxBitrate     *int     `json:"maxbitrate"`
+	PlaybackMethod []int    `json:"playbackmethod"`
+	PlaybackEnd    *int     `json:"playbackend"`
+	Delivery       []int    `json:"delivery"`
+	Pos            *int     `json:"pos"`
+	API            []int    `json:"api"`
+}
+
+type BadInput struct {
+	Message string
+}
+
+func (e *BadInput) Error() string {
+	return e.Message
+}
+
+// Validation functions
+func isInteger(value interface{}) bool {
+	switch v := value.(type) {
+	case int:
+		return true
+	case float64:
+		return v == float64(int(v))
+	case string:
+		_, err := strconv.Atoi(v)
+		return err == nil
+	default:
+		return false
+	}
+}
+
+func isArrayOfNums(value interface{}) bool {
+	switch v := value.(type) {
+	case []int:
+		return true
+	case []float64:
+		for _, num := range v {
+			if num != float64(int(num)) {
+				return false
+			}
+		}
+		return true
+	case []string:
+		for _, str := range v {
+			if _, err := strconv.Atoi(str); err != nil {
+				return false
+			}
+		}
+		return true
+	default:
+		return false
+	}
+}
+
+// Define valid values
+var validLinearity = map[int]bool{1: true}
+var validSkip = map[int]bool{0: true, 1: true}
+var validPlaybackEnd = map[int]bool{1: true, 2: true, 3: true}
+var validPos = map[int]bool{0: true, 1: true, 2: true, 3: true, 4: true, 5: true, 6: true, 7: true}
+
+// Map parameters to validation functions
+var optionalVideoParams = map[string]func(interface{}) bool{
+	"minduration":    isInteger,
+	"maxduration":    isInteger,
+	"protocols":      isArrayOfNums,
+	"startdelay":     isInteger,
+	"linearity":      func(value interface{}) bool { return isInteger(value) && validLinearity[toInt(value)] },
+	"skip":           func(value interface{}) bool { return isInteger(value) && validSkip[toInt(value)] },
+	"skipmin":        isInteger,
+	"skipafter":      isInteger,
+	"sequence":       isInteger,
+	"battr":          isArrayOfNums,
+	"maxextended":    isInteger,
+	"minbitrate":     isInteger,
+	"maxbitrate":     isInteger,
+	"playbackmethod": isArrayOfNums,
+	"playbackend":    func(value interface{}) bool { return isInteger(value) && validPlaybackEnd[toInt(value)] },
+	"delivery":       isArrayOfNums,
+	"pos":            func(value interface{}) bool { return isInteger(value) && validPos[toInt(value)] },
+	"api":            isArrayOfNums,
+}
+
+// Helper function to convert interface to int
+func toInt(value interface{}) int {
+	switch v := value.(type) {
+	case int:
+		return v
+	case float64:
+		return int(v)
+	case string:
+		if i, err := strconv.Atoi(v); err == nil {
+			return i
+		}
+	}
+	return 0
+}
```

Comment:
Based on https://github.com/prebid/prebid-server/pull/3806/files#r1705364786 comment this function is not required.